### ISSUE TITLE
Error when server returns status of 204

### ISSUE THREAD

**Issue by NeoPhi** [bitinn/node-fetch#115]
When making a POST request against a server that returns a 204 status code the following error occurs with node v6.1.0 and node-fetch v1.5.2:

```
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: unexpected end of file
    at Zlib._handle.onerror (zlib.js:370:17)
```

**Comment by bitinn** [comment]
I will make a test case for it soon, probably because there is no content to decode, not sure how node.js handle pipeing in this case yet.

**Comment by jimmywarting** [comment]
I think this is related to my caching problem where i get `304 not modified` and a empty response

**Comment by NeoPhi** [comment]
This only seems to be a problem if the server returns a `Content-Encoding: gzip` header as part of the 204 response. If I pass the `compress: false` option to the call, no error is generated.

**Comment by bitinn** [comment]
this is now fixed and released as v1.5.3

**Comment by artola** [comment]
I got the same error for HEAD request with 404 status, unless pass `compress: false` ... while v1.5.3 (b46c5ea9ee6692676a2824334e24f9d3b95d3b78)
